cmd/bosun/sched/check.go

```diff
@@ -140,6 +135,61 @@ func (s *Schedule) runHistory(r *RunHistory, ak models.AlertKey, event *models.E
 			}
 		}
 	}()
+	if incident != nil {
+		rt, err = data.GetRenderedTemplates(incident.Id)
+		if err != nil {
+			return
+		}
+		for i, action := range incident.Actions {
+			if action.Type == models.ActionDelayedClose && !(action.Fullfilled || action.Cancelled) {
+				if event.Status > incident.WorstStatus {
+					// If the lifetime severity of the incident has increased, cancel the delayed close
+					cerr := s.ActionByAlertKey("bosun", "cancelled delayed close due to severity increase", models.ActionCancelClose, nil, ak)
+					incident, err = data.GetIncidentState(incident.Id)
+					if cerr != nil {
+						slog.Errorln(cerr)
+					}
+					// Continue processing alert after cancelling the delayed close
+					break
+				}
+				if action.Deadline == nil {
```

Comment:
If this should never happen, lets not bog down runHistory with another if statement.